Source/Fuse.Scripting.JavaScript/Tests/Model/Model.Test.uno

```diff
@@ -0,0 +1,581 @@
+				Assert.AreEqual( 116, e.v.Value );
+				Assert.AreEqual( 16, e.r.Value );
+				Assert.AreEqual( 232, e.d.Value );
+			}
+		}
+
+		[Test]
+		//TODO: working on this this one still (found the Each one)
+		public void UseCase1()
+		{
+			var e = new UX.Model.UseCase1();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+
+				Assert.AreEqual( "one,two,three,four,five", GetRecursiveText(e.a) );
+				Assert.AreEqual( "five", GetText(e.s) );
+
+				var two = e.a.FindNodeByName( "two" ) as UMUItem;
+				two.callAdd.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( "five,two", GetText(e.s) );
+
+				var five = e.a.FindNodeByName( "five" ) as UMUItem;
+				five.callRemove.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( "two", GetText(e.s) );
+			}
+		}
+
+		[Test]
+		public void Each()
+		{
+			var e = new UX.Model.Each();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+
+				Assert.AreEqual( "three,two,one", GetDudZ(e.a));
+				Assert.AreEqual( "three,two,one", GetDudZ(e.b));
+			}
+		}
+
+		[Test]
+		public void NestedAccessor()
+		{
+			var e = new UX.Model.NestedAccessor();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( 110, e.v.Value );
+				Assert.AreEqual( 10, e.r.Value );
+				Assert.AreEqual( -10, e.d.Value );
+
+				e.v.Value = 115;
+				root.StepFrameJS();
+				Assert.AreEqual( 115, e.v.Value );
+				Assert.AreEqual( 15, e.r.Value );
+				Assert.AreEqual( -15, e.d.Value );
+
+				e.callIncr.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual( 116, e.v.Value );
+				Assert.AreEqual( 16, e.r.Value );
+				Assert.AreEqual( -16, e.d.Value );
+			}
+		}
+
+		[Test]
+		public void Test1()
+		{
+			var e = new UX.Model.Test1();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual(false, e.mySwitch.Value);
+				Assert.AreEqual(true, e.myFlippedSwitch.Value);
+
+				e.mySwitch.Value = true;
+
+				Assert.AreEqual(true, e.mySwitch.Value);
+				root.StepFrameJS();
+				Assert.AreEqual(true, e.mySwitch.Value);
+				Assert.AreEqual(false, e.myFlippedSwitch.Value);
+
+				e.myFlippedSwitch.Value = true;
+				Assert.AreEqual(true, e.myFlippedSwitch.Value);
+				root.StepFrameJS();
+
+				Assert.AreEqual(false, e.mySwitch.Value);
+				Assert.AreEqual(true, e.myFlippedSwitch.Value);
+
+				e.myFlippedSwitch.Value = false;
+				e.mySwitch.Value = true;
+				e.myFlippedSwitch.Value = true;
+				root.StepFrameJS();
+
+				Assert.AreEqual(false, e.mySwitch.Value);
+				Assert.AreEqual(true, e.myFlippedSwitch.Value);
+
+			}
+		}
+
+		[Test]
+		public void ListOrder()
+		{
+			var e = new UX.Model.ListOrder();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				Assert.AreEqual( "2,3", e.oc.JoinValues() );
+				e.oc.Log.Clear(); //TODO: it's undecided what the bootstrapping messages will be still
```

Comment:
Sounds like there's some more work to be done here. Either fix it, or file a ticket (and put a reference to the ticket here)...